src/js.cc

```diff
@@ -5755,16 +5766,74 @@ js_get_value_date(js_env_t *env, js_value_t *value, double *result) {
 }
 
 extern "C" int
-js_get_array_length(js_env_t *env, js_value_t *value, uint32_t *result) {
+js_get_array_length(js_env_t *env, js_value_t *array, uint32_t *result) {
   // Allow continuing even with a pending exception
 
-  auto local = js_to_local<Array>(value);
+  auto local = js_to_local<Array>(array);
 
   *result = local->Length();
 
   return 0;
 }
 
+extern "C" int
+js_get_array_elements(js_env_t *env, js_value_t *array, js_value_t **elements, size_t len, size_t offset, uint32_t *result) {
+  if (env->is_exception_pending()) return js_error(env);
+
+  auto context = env->current_context();
+
+  auto local = js_to_local<Array>(array);
+
+  uint32_t written = 0;
+
+  auto success = env->call_into_javascript<bool>(
+    [&] {
+      for (uint32_t i = 0, n = len, j = offset, m = local->Length(); i < n && j < m; i++, j++) {
+        auto value = local->Get(context, j);
+
+        if (value.IsEmpty()) return false;
```

Comment:
Does this check abort on sparse arrays? `[0, 1, 2, undefined, undefined, 4]` ?

Reply:
No, it checks for exceptions as the heap handle is only empty if an exception occurred.

Reply:
Aha! thanks for clarifying; I was wondering if the logic should be mirrored in [bare-napi-compat#1](https://github.com/holepunchto/bare-compat-napi/pull/1)
but it already is through napi `status` abort:

https://github.com/nodejs/node/blob/8342183036400b32dc2f51f465b99e9f142adf20/src/js_native_api_v8.cc#L1382
https://github.com/nodejs/node/blob/8342183036400b32dc2f51f465b99e9f142adf20/src/js_native_api_v8.h#L293

thanks!